ccmlib/scylla_node.py

```diff
@@ -536,51 +537,69 @@ def start(self, join_ring=True, no_wait=False, verbose=False,
         jvm_args = jvm_args + ['--collectd-hostname',
                                '%s.%s' % (socket.gethostname(), self.name)]
 
-        # Let's add jvm_args and the translated args
-
-        args = [launch_bin, '--options-file', options_file, '--log-to-stdout', '1'] + jvm_args + translated_args
+        args = [launch_bin, '--options-file', options_file, '--log-to-stdout', '1']
+
+        # holds all extended args except --smp and --memory
+        # that are handled separately
+        ext_args = OrderedDict()
+
+        MB = 1024*1024
+
+        def process_opts(opts, set_during_test = False):
+            opts_i = 0
+            memory_opt = None
+            while opts_i < len(opts):
+                if opts[opts_i].startswith("--scylla-manager="):
+                    opts_i += 1
+                elif opts[opts_i].startswith('-'):
```

Comment:
nit, could put something like:

```suggestion
                else:
                    # the command line options shows up either like "--foo value-of-foo"
                    # or as a single option like --yes-i-insist
                    assert(opts[opts_i].startswith('-')
```

otherwise human reader would think there could be another option that we will end up in an infinite loop, as the nonexistent else branch does not increase `opts_i`.

Reply:
good idea

Reply:
done in v2 (just earlier in the code)